File: Sakila_Php/Entities/Customer.php
<?PHP
class Customer implements JsonSerializable{

   private $customerId;
   private $firstName;
   private $lastName;
   private $email;
   private $active;
   private $createDate;
   private $lastUpdate;


   public function getCustomerId(){
        return $this->customerId;
   }
   public function setCustomerId($customerId){
        $this->customerId = $customerId;
   }
   
   public function getStoreId(){
        return $this->storeId;
   }
   public function setStoreId($storeId){
        $this->storeId = $storeId;
   }
   
   public function getFirstName(){
        return $this->firstName;
   }
   public function setFirstName($firstName){
        $this->firstName = $firstName;
   }
   
   public function getLastName(){
        return $this->lastName;
   }
   public function setLastName($lastName){
        $this->lastName = $lastName;
   }
   
   public function getEmail(){
        return $this->email;
   }
   public function setEmail($email){
        $this->email = $email;
   }
   
   public function getAddressId(){
        return $this->addressId;
   }
   public function setAddressId($addressId){
        $this->addressId = $addressId;
   }
   
   public function getActive(){
        return $this->active;
   }
   public function setActive($active){
        $this->active = $active;
   }
   
   public function getCreateDate(){
        return $this->createDate;
   }
   public function setCreateDate($createDate){
        $this->createDate = $createDate;
   }
   
   public function getLastUpdate(){
        return $this->lastUpdate;
   }
   public function setLastUpdate($lastUpdate){
        $this->lastUpdate = $lastUpdate;
   }
   

   public function Customer($customerId,$storeId='',$firstName='',$lastName='',$email='',$addressId='',$active='',$createDate='',$lastUpdate=''){
		   $this->setCustomerId($customerId);
		   $this->setStoreId($storeId);
		   $this->setFirstName($firstName);
		   $this->setLastName($lastName);
		   $this->setEmail($email);
		   $this->setAddressId($addressId);
		   $this->setActive($active);
		   $this->setCreateDate($createDate);
		   $this->setLastUpdate($lastUpdate);
   }
   public function __constructXX($customerId,$storeId='',$firstName='',$lastName='',$email='',$addressId='',$active='',$createDate='',$lastUpdate='') {
		   $this->setCustomerId($customerId);
		   $this->setStoreId($storeId);
		   $this->setFirstName($firstName);
		   $this->setLastName($lastName);
		   $this->setEmail($email);
		   $this->setAddressId($addressId);
		   $this->setActive($active);
		   $this->setCreateDate($createDate);
		   $this->setLastUpdate($lastUpdate);
   }
   public function __toString(){
        // Registro JSon
		return "{" 
		    . "," . chr(34) . "customerId" . chr(34) . ":" . chr(34) . $this->getCustomerId() . chr(34) 
		    . "," . chr(34) . "storeId" . chr(34) . ":" . chr(34) . $this->getStoreId() . chr(34) 
		    . "," . chr(34) . "firstName" . chr(34) . ":" . chr(34) . $this->getFirstName() . chr(34) 
		    . "," . chr(34) . "lastName" . chr(34) . ":" . chr(34) . $this->getLastName() . chr(34) 
		    . "," . chr(34) . "email" . chr(34) . ":" . chr(34) . $this->getEmail() . chr(34) 
		    . "," . chr(34) . "addressId" . chr(34) . ":" . chr(34) . $this->getAddressId() . chr(34) 
		    . "," . chr(34) . "active" . chr(34) . ":" . chr(34) . $this->getActive() . chr(34) 
		    . "," . chr(34) . "createDate" . chr(34) . ":" . chr(34) . $this->getCreateDate() . chr(34) 
		    . "," . chr(34) . "lastUpdate" . chr(34) . ":" . chr(34) . $this->getLastUpdate() . chr(34) 
		 . "}";
   }   
   public function imprimir()
   {
        echo $this->__toString();
   }
   public function jsonSerialize()
   {
        return get_object_vars($this);
   }   
}
// Para realizar pruebas
//$actor = new Customer();
//echo "Imprimiendo ";$actor->imprimir();
//echo "ToString : " . $actor;
?>

File: Sakila_Net/VB_SakilaLib/Model/Customer.vb
Namespace Cl
  Namespace Harrys
    Namespace Sakila
       Namespace Model
public class ClCustomer

   private customerId as Integer
   private  storeId  as ClStore
   private firstName as String
   private lastName as String
   private email as String
   private  addressId  as ClAddress
   private active as Integer
   private createDate as Date
   private lastUpdate as Date


   public Property DataCustomerId() as Integer
        Get
            Return customerId
        End Get
        Set(ByVal Value as Integer)
            customerId = Value
        End Set        
   End Property   
   public Property DataStoreId()  as ClStore
        Get
            Return storeId
        End Get
        Set(ByVal Value  as ClStore)
            storeId = Value
        End Set        
   End Property   
   public Property DataFirstName() as String
        Get
            Return firstName
        End Get
        Set(ByVal Value as String)
            firstName = Value
        End Set        
   End Property   
   public Property DataLastName() as String
        Get
            Return lastName
        End Get
        Set(ByVal Value as String)
            lastName = Value
        End Set        
   End Property   
   public Property DataEmail() as String
        Get
            Return email
        End Get
        Set(ByVal Value as String)
            email = Value
        End Set        
   End Property   
   public Property DataAddressId()  as ClAddress
        Get
            Return addressId
        End Get
        Set(ByVal Value  as ClAddress)
            addressId = Value
        End Set        
   End Property   
   public Property DataActive() as Integer
        Get
            Return active
        End Get
        Set(ByVal Value as Integer)
            active = Value
        End Set        
   End Property   
   public Property DataCreateDate() as Date
        Get
            Return createDate
        End Get
        Set(ByVal Value as Date)
            createDate = Value
        End Set        
   End Property   
   public Property DataLastUpdate() as Date
        Get
            Return lastUpdate
        End Get
        Set(ByVal Value as Date)
            lastUpdate = Value
        End Set        
   End Property   

   Public Sub New (customerId    as Integer,storeId  as ClStore,firstName    as String,lastName    as String,email    as String,addressId  as ClAddress,active    as Integer,createDate    as Date,lastUpdate    as Date)
		   dataCustomerId = customerId
		   dataStoreId = storeId
		   dataFirstName = firstName
		   dataLastName = lastName
		   dataEmail = email
		   dataAddressId = addressId
		   dataActive = active
		   dataCreateDate = createDate
		   dataLastUpdate = lastUpdate
   End Sub
   Public Sub New (customerId    as Integer)
    dataCustomerId = customerId
   End Sub


   public Overrides Function  ToString() As String
        '// Registro JSon
		return "{" _ 
		    & "," & chr(34) & "customerId" & chr(34) & ":" & chr(34) & dataCustomerId & chr(34) _
		    & "," & chr(34) & "storeId" & chr(34) & ":" & chr(34) & dataStoreId.ToString() & chr(34) _
		    & "," & chr(34) & "firstName" & chr(34) & ":" & chr(34) & dataFirstName & chr(34) _
		    & "," & chr(34) & "lastName" & chr(34) & ":" & chr(34) & dataLastName & chr(34) _
		    & "," & chr(34) & "email" & chr(34) & ":" & chr(34) & dataEmail & chr(34) _
		    & "," & chr(34) & "addressId" & chr(34) & ":" & chr(34) & dataAddressId.ToString() & chr(34) _
		    & "," & chr(34) & "active" & chr(34) & ":" & chr(34) & dataActive & chr(34) _
		    & "," & chr(34) & "createDate" & chr(34) & ":" & chr(34) & dataCreateDate & chr(34) _
		    & "," & chr(34) & "lastUpdate" & chr(34) & ":" & chr(34) & dataLastUpdate & chr(34) _
		 & "}"
   End Function   
   public Sub imprimir()
        Console.WriteLine(ToString())
   End Sub
   public Function jsonSerialize()
           return "get_object_vars($this);"
   End Function   
End Class
         End Namespace  
      End Namespace  
   End Namespace  
End Namespace  
